addons/website/static/src/snippets/s_embed_code/options.js

```diff
@@ -15,71 +34,18 @@ options.registry.EmbedCode = options.Class.extend({
         const $container = this.$target.find('.s_embed_code_embedded');
         const code = $container.html().trim();
 
-        await loadBundle({
-            jsLibs: [
-                '/web/static/lib/ace/ace.js',
-                '/web/static/lib/ace/mode-xml.js',
-                '/web/static/lib/ace/mode-qweb.js',
-            ],
-        });
-
         await new Promise(resolve => {
-            const $content = $(renderToElement('website.custom_code_dialog_content'));
-            const aceEditor = this._renderAceEditor($content.find('.o_ace_editor_container')[0], code || '');
-            const dialog = new Dialog(this, {
+            this.dialog.add(CodeEditorDialog, {
                 title: _t("Edit embedded code"),
-                $content,
-                buttons: [
-                    {
-                        text: _t("Save"),
-                        classes: 'btn-primary',
-                        click: async () => {
-                            $container[0].innerHTML = aceEditor.getValue();
-                        },
-                        close: true,
-                    },
-                    {
-                        text: _t("Discard"),
-                        close: true,
-                    },
-                ],
+                value: code,
+                mode: "xml",
+                confirm: (newValue) => {
+                   $container[0].innerHTML = newValue;
```

Comment:
isn't the goal of the embedded code block to display source code? because if you write HTML in it, it will evaluate it